docs/overview/threat-model.md

```diff
@@ -0,0 +1,83 @@
+
+## Integrator Responsibilities
+
+As an integrator of Backstage you yourself are responsible for protecting your Backstage installation from external and unauthorized access. The sign-in system in Backstage does not exist to limit access, only to inform the system of the identity of the user. There are some plugins that have more fine-grained access control through the permissions system, but the primary purpose of that system is to restrict access to resources for internal users rather than Backstage as a whole. A common and recommended way to protect a Backstage deployment from unauthorized access is to deploy it behind an authenticating proxy such as AWS’s ALB, GCP’s IAP, or Cloudflare Access.
+
+Other responsibilities include protecting the integrity of configuration files as it may otherwise be possible to introduce vulnerable configurations, as well as the confidentiality of configured secrets related to Backstage as these typically include authentication details to third party systems.
+
+The integrator is ultimately responsible for auditing usage of internal and external plugins as these run on the host system and have access to configuration and secrets. When installing plugins from sources like NPM, you should vet these in the same way that you would vet any other package installed from that source.
+
+## Common Backend Configuration
+
+There are many common facilities that are configured centrally and available to all Backstage backend plugins. For example there is a `DatabaseManager` that provides access to a SQL database, `TaskScheduler` for scheduling long-running tasks, `Logger` as a general logging facility, and `UrlReader` for reading content from external sources. These are all configured either directly in code, or within the `backend` block of the static configuration. The appropriate care needs to be taken to ensure that any secrets remain confidential and no malicious configuration is injected.
+
+In a typical Backstage setup, there is no boundary between plugins that run on the same host. Likewise, there is no boundary between plugins that share the same database. Any plugin that is running on a host that has access to the database of any other plugin should be considered to have full access to that other plugin. For example, if the `catalog` plugin is deployed adjacent to the `auth` plugin and it has access to its database, the `catalog` can be considered to have full access to the `auth` plugin, even if they do not run on the same host. The only way to create a boundary between them is to deploy them in such a way that they do not have access to each others’ database. This applies to the database facility as well as any other shared resources, such as the cache.
```

Comment:
"it has access to its database" is perhaps a bit vague in terms of who "it" is respectively, but it's a very small nit

Reply:
The wording here doesn't specify whether it's talking about logical or physical databases that it's referring to, or what kind of hardening that it would entail if one were to want to avoid the mentioned situation. I realise that it's mainly meant as an example, but just wanted to note it anyway

Reply:
Agreed 👍, tried to reword it a bit